yarn-project/simulator/src/public/avm_executor.test.ts

```diff
@@ -0,0 +1,83 @@
+import { AztecAddress, CallContext, EthAddress, FunctionData, FunctionSelector, Header } from '@aztec/circuits.js';
+import { makeHeader } from '@aztec/circuits.js/testing';
+import { Fr } from '@aztec/foundation/fields';
+import { AvmTestContractArtifact } from '@aztec/noir-contracts.js';
+
+import { MockProxy, mock } from 'jest-mock-extended';
+
+import { CommitmentsDB, PublicContractsDB, PublicStateDB } from './db.js';
+import { PublicExecution } from './execution.js';
+import { PublicExecutor } from './executor.js';
+
+/*
+ * These tests are very flaky at the moment as the dependency on the bb binary
+ * is not working when running on the CI. To get around this we trivially pass if we fail to spawn the binary.
+ * This is not ideal and should be fixed in the future. But the test will still fail if the binary is found but the proof is invalid.
+ */
+describe('AVM WitGen and Proof Generation', () => {
+  let publicState: MockProxy<PublicStateDB>;
+  let publicContracts: MockProxy<PublicContractsDB>;
+  let commitmentsDb: MockProxy<CommitmentsDB>;
+  let header: Header;
+
+  const callContext = CallContext.from({
+    msgSender: AztecAddress.random(),
+    storageContractAddress: AztecAddress.random(),
+    portalContractAddress: EthAddress.random(),
+    functionSelector: FunctionSelector.empty(),
+    isContractDeployment: false,
+    isDelegateCall: false,
+    isStaticCall: false,
+    startSideEffectCounter: 0,
+  });
+  const contractAddress = AztecAddress.random();
+
+  beforeEach(() => {
+    publicState = mock<PublicStateDB>();
+    publicContracts = mock<PublicContractsDB>();
+    commitmentsDb = mock<CommitmentsDB>();
+
+    const randomInt = Math.floor(Math.random() * 1000000);
+    header = makeHeader(randomInt);
+  }, 10000);
+
+  it('Should prove valid execution of bytecode that performs addition', async () => {
+    const args: Fr[] = [new Fr(1), new Fr(2)];
+    // Bytecode for the following contract is encoded:
+    // const bytecode = encodeToBytecode([
+    //    new CalldataCopy(/*indirect=*/ 0, /*cdOffset=*/ 0, /*copySize=*/ 2, /*dstOffset=*/ 0),
+    //    new Add(/*indirect=*/ 0, TypeTag.FIELD, /*aOffset=*/ 0, /*bOffset=*/ 1, /*dstOffset=*/ 2),
+    //    new Return(/*indirect=*/ 0, /*returnOffset=*/ 2, /*copySize=*/ 1),
+    // ]);
+    const bytecode: Buffer = Buffer.from('HwAAAAAAAAAAAgAAAAAAAAYAAAAAAAAAAQAAAAI3AAAAAAIAAAAB', 'base64');
```

Comment:
I would rather use HEX encoding. The goal is not to save space but make it readable for the developer.

Reply:
> Do you think we should we create this from an opcode array then call serialize; rather than hard coding the string?

He might be trying to avoid introducing a dependency to the instructions and the simulator. I think that's right because at this level of abstraction that shouldn't matter. However, having the hardcoded bytecode is also a pain if we ever change anything.

I'd suggest just removing this test and relying on the one(s) using "AvmTestContractArtifact."

Reply:
Yep i was hesitatnt to introduce the dependency on the simulator here.

The plan would be to use the one from the `AvmTestContractArtifact` when the AVM supports the MOV opcode(it's skipped below this test until then).

Once that is supported I'll happily remove this test